sql/core/src/main/scala/org/apache/spark/sql/execution/basicOperators.scala

```diff
@@ -306,6 +306,8 @@ case class UnsafeExternalSort(
   override def output: Seq[Attribute] = child.output
 
   override def outputOrdering: Seq[SortOrder] = sortOrder
+
+  override def outputsUnsafeRows: Boolean = true
```

Comment:
or actually, maybe we can add a general assertion to the `execute` method of `SparkPlan`?

`assert(children.map(_.outputsUnsafeRows).distinct <= 1)`